src/command/command.rs

```diff
@@ -64,6 +66,35 @@ impl Command {
                 }
                 Ok((Command::Command, 2))
             }
+            _ if cmd == "GET" => {
+                if length != 2 {
+                    return Err(CommandError::Malformed);
+                }
+                let key = match &tokens[2] {
+                    Token::SimpleString(s) => s.bytes().collect(),
+                    Token::BulkString(Some(s)) => s.clone(),
+                    _ => return Err(CommandError::Malformed),
+                };
+
+                Ok((Command::Get(key), 3))
+            }
+            _ if cmd == "SET" => {
+                if length != 3 {
+                    return Err(CommandError::Malformed);
+                }
+                let key = match &tokens[2] {
```

Comment:
Indexing to an undefined position will panic. You can instead `match tokens.get(2)` for `Some(x)` rather than matching `tokens[2]` for `x`.

Reply:
Good call. I was putting off refactoring this code, but got through it in a pairing session and I think it's much better now, including this suggestion.